server/configs/config.js

```diff
@@ -5,16 +5,48 @@ dotenv.config();
 const config = {
   development: {
     url: process.env.DEV_DATABASE_URL,
-    dialect: process.env.DEV_DIALECT
+    dialect: process.env.DEV_DIALECT,
   },
   test: {
     dialect: process.env.TEST_DIALECT,
-    storage: process.env.TEST_STORAGE
+    storage: process.env.TEST_STORAGE,
   },
   production: {
     url: process.env.PROD_DATABASE_URL,
-    dialect: process.env.PROD_DIALECT
-  }
+    dialect: process.env.PROD_DIALECT,
+  },
 };
 
 export default config;
+const isProd = process.env.NODE_ENV === "production";
+
+export const socialAuthConfig = {
+  facebook: isProd
+    ? {
+        clientID: process.env.FACEBOOK_CLIENT_ID_PROD,
+        clientSecret: process.env.FACEBOOK_CLIENT_SECRET_PROD,
+        callbackURL: process.env.FB_REDIRECT_PROD,
+      }
+    : {
+        clientID: process.env.FACEBOOK_CLIENT_ID,
+        clientSecret: process.env.FACEBOOK_CLIENT_SECRET,
+        callbackURL: `http://localhost:${
+          process.env.PORT
+        }/api/v1/auth/facebook/redirect`,
+      },
+  google: {
+    clientID: process.env.GOOGLE_CLIENT_ID_1,
+    clientSecret: process.env.GOOGLE_CLIENT_SECRET_1,
+    callbackURL: isProd
+      ? process.env.G_REDIRECT_PROD
+      : `http://localhost:${process.env.PORT}/api/v1/auth/google/redirect`,
+  },
+
+  twitter: {
+    consumerKey: process.env.TWITTER_CONSUMER_KEY,
+    consumerSecret: process.env.TWITTER_CONSUMER_SECRET,
+    callbackURL: isProd
+      ? process.env.TW_REDIRECT_PROD
+      : `http://localhost:${process.env.PORT}/api/v1/auth/twitter/redirect`,
```

Comment:
This file is barely readable and loads of functionality is repeated. Let us keep this DRY